front/pages/api/w/[wId]/data_sources/[name]/managed/permissions.ts

```diff
@@ -0,0 +1,96 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSource } from "@app/lib/api/data_sources";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { ConnectorResource, ConnectorsAPI } from "@app/lib/connectors_api";
+import { ReturnedAPIErrorType } from "@app/lib/error";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+export type GetDataSourcePermissionsResponseBody = {
+  resources: ConnectorResource[];
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    GetDataSourcePermissionsResponseBody | ReturnedAPIErrorType | void
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
```

Comment:
copy-paste mistake or intentional? I saw "workspace not found in some other api calls

Reply:
this is just a reorder we do use `data_source_not_found` :+1: